cypress/utils/apps/query_enhancements/inspect.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { QueryLanguages } from './constants';
+
+const visualizationTitlesWithNoInspectOptions = [
+  '[Flights]Controls',
+  '[Flights]MarkdownInstructions',
+  '[Flights]Delays&Cancellations',
+];
+
+const visualizationTitlesWithInspectOptions = [
+  '[Flights]AirlineCarrier',
+  '[Flights]FlightCountandAverageTicketPrice',
+  '[Flights]TotalFlights',
+  '[Flights]AverageTicketPrice',
+];
+
+/**
+ * Gets Date and returns string of date in format Jan 24, 2025 @ 16:20:08.000
+ * @param {Date} date - date to format
+ * @returns {string} - in the format e.g. Jan 24, 2025 @ 16:20:08.000
+ */
+function formatDate(date) {
+  const months = [
+    'Jan',
+    'Feb',
+    'Mar',
+    'Apr',
+    'May',
+    'Jun',
+    'Jul',
+    'Aug',
+    'Sep',
+    'Oct',
+    'Nov',
+    'Dec',
+  ];
+
+  const month = months[date.getMonth()];
+  const day = date.getDate().toString();
+  const year = date.getFullYear();
+  const hours = date.getHours().toString().padStart(2, '0');
+  const minutes = date.getMinutes().toString().padStart(2, '0');
+  const seconds = date.getSeconds().toString().padStart(2, '0');
+  const milliseconds = date.getMilliseconds().toString().padStart(3, '0');
+
+  return `${month} ${day}, ${year} @ ${hours}:${minutes}:${seconds}.${milliseconds}`;
+}
+
+/**
+ * Format value to the corresponding string expected in the UI.
```

Comment:
```suggestion
/**
 * Formats values for UI display by:
 * - Adding thousand separators to numbers using locale settings
 * - Converting SQL/PPL datetime strings (YYYY-MM-DD HH:MM:SS.SSS) to formatted dates
 * - Passing through other values unchanged
 * @param {string|number} value - The value to format
 * @returns {string} The formatted value
 * @example
 * formatValue(1000) // "1,000"
 * formatValue("2025-01-24 16:20:08.000") // "Jan 24, 2025 @ 16:20:08.000"
 */
```

Reply:
nit: It would be great if you could have separate functions for number, string and date, like formatNumber, formatString and formatDate. Then in the formatValue you call them. I think in sharing or some other tests, we also need to format date. So it would be great to have separate format helpers.